stan/math/rev/mat/functor/cvodes_integrator.hpp

```diff
@@ -99,24 +105,20 @@ class cvodes_integrator {
     typedef stan::is_var<T_initial> initial_var;
     typedef stan::is_var<T_param> param_var;
 
-    check_finite("integrate_ode_cvodes", "initial state", y0);
-    check_finite("integrate_ode_cvodes", "initial time", t0);
-    check_finite("integrate_ode_cvodes", "times", ts);
-    check_finite("integrate_ode_cvodes", "parameter vector", theta);
-    check_finite("integrate_ode_cvodes", "continuous data", x);
-    check_nonzero_size("integrate_ode_cvodes", "times", ts);
-    check_nonzero_size("integrate_ode_cvodes", "initial state", y0);
-    check_ordered("integrate_ode_cvodes", "times", ts);
-    check_less("integrate_ode_cvodes", "initial time", t0, ts[0]);
-    if (relative_tolerance <= 0)
-      invalid_argument("integrate_ode_cvodes", "relative_tolerance,",
-                       relative_tolerance, "", ", must be greater than 0");
-    if (absolute_tolerance <= 0)
-      invalid_argument("integrate_ode_cvodes", "absolute_tolerance,",
-                       absolute_tolerance, "", ", must be greater than 0");
-    if (max_num_steps <= 0)
-      invalid_argument("integrate_ode_cvodes", "max_num_steps,", max_num_steps,
-                       "", ", must be greater than 0");
+    const char* fun = "integrate_ode_cvodes";
+
+    check_finite(fun, "initial state", y0);
+    check_finite(fun, "initial time", t0);
+    check_finite(fun, "times", ts);
+    check_finite(fun, "parameter vector", theta);
+    check_finite(fun, "continuous data", x);
+    check_nonzero_size(fun, "times", ts);
+    check_nonzero_size(fun, "initial state", y0);
+    check_ordered(fun, "times", ts);
+    check_less(fun, "initial time", t0, ts[0]);
+    check_positive(fun, "relative tolerance", relative_tolerance);
```

Comment:
These should be changed back to the old versions. We want it to throw `invalid_argument` instead of `domain_error`. That's for the checks on `relative_tolerance`, `absolute_tolerance`, and `max_num_steps`.

Reply:
I missed the difference with `invalid_argument`.  

What's the general principle for deciding when to throw which exception?  I thought we were only throwing the `invalid_argument` for indexing errors.  These don't look like indexing.  So why wouldn't, say he check on time ordering also throw `invalid_argument`?

Reply:
It could be a different exception, just not `domain_error` or an exception derived from that class. 

`domain_error`s are caught and treated as recoverable exceptions. From NUTS, it's treated as a rejection. There's a chance that the next set of inputs to the log prob function may be computed fine. 

Here, if the relative tolerance is set to a negative value, it was configured improperly and we do want everything to stop. We don't have to: if it was a `domain_error` Stan would run for 100 iterations, spit out rejection messages, then stop. I think that's not ideal. 

And yes, the check on time should also throw a non `domain_error`.